src/x25519.rs

```diff
@@ -76,14 +76,19 @@ impl EphemeralSecret {
         SharedSecret(self.0 * their_public.0)
     }
 
-    /// Generate an x25519 [`EphemeralSecret`] key.
-    pub fn new<T: RngCore + CryptoRng>(mut csprng: T) -> Self {
```

Comment:
It might be good to retain `new` with a `#[deprecated]` attribute to help people upgrade

Reply:
Good idea. btw - should we take random* out from `StaticSecret` as I've seen people using it just to ECDH when they should be using `EphemeralSecret` really.

Reply:
Done - I've added `#[deprecated]` agains 2.0.0 with indication of removal in 2.1.0